cmd/unpack/main.go

```diff
@@ -24,36 +39,99 @@ func main() {
 				fmt.Printf("Git commit: %s\n", version.String())
 				os.Exit(0)
 			}
-			bundleFS := os.DirFS(bundleDir)
-			bundleContents := map[string][]byte{}
-			if err := fs.WalkDir(bundleFS, ".", func(path string, d fs.DirEntry, err error) error {
-				if err != nil {
-					return err
-				}
-				if d.IsDir() {
-					return nil
-				}
-				data, err := fs.ReadFile(bundleFS, path)
-				if err != nil {
-					return fmt.Errorf("read file %q: %w", path, err)
-				}
-				bundleContents[path] = data
-				return nil
-			}); err != nil {
-				log.Fatalf("walk bundle dir %q: %v", bundleDir, err)
+			contentHandler := serveBundle(os.DirFS(bundleDir))
+			h := mux.NewRouter()
+			h.Handle("/content", contentHandler).Methods(http.MethodGet)
+			srv := http.Server{
+				Addr:         listenAddr,
+				Handler:      handlers.CombinedLoggingHandler(os.Stdout, h),
+				ReadTimeout:  1 * time.Second,
+				WriteTimeout: 10 * time.Second,
 			}
+			go func() {
+				log.Println("listening on", srv.Addr)
+				if err := srv.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
+					log.Fatal(err)
+				}
+			}()
 
-			encoder := json.NewEncoder(os.Stdout)
-			if err := encoder.Encode(bundleContents); err != nil {
+			ctx, cancel := signal.NotifyContext(cmd.Context(), syscall.SIGINT, syscall.SIGTERM)
+			defer cancel()
+			<-ctx.Done()
+
+			shutdownCtx, cancelShutdown := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancelShutdown()
+			log.Println("shutting down")
+			if err := srv.Shutdown(shutdownCtx); err != nil {
 				log.Fatal(err)
 			}
+			log.Println("shutdown complete")
+
 			return nil
 		},
 	}
 	cmd.Flags().StringVar(&bundleDir, "bundle-dir", "", "directory in which the bundle can be found")
+	cmd.Flags().StringVar(&listenAddr, "listen-addr", ":8080", "listen address for http server")
 	cmd.Flags().BoolVar(&rukpakVersion, "version", false, "displays rukpak version information")
 
 	if err := cmd.Execute(); err != nil {
 		log.Fatal(err)
 	}
 }
+
+func serveBundle(bundle fs.FS) http.HandlerFunc {
+	return func(writer http.ResponseWriter, request *http.Request) {
+		buf := &bytes.Buffer{}
+		gzw := gzip.NewWriter(buf)
+		tw := tar.NewWriter(gzw)
+		if err := fs.WalkDir(bundle, ".", func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if d.Type()&os.ModeSymlink != 0 {
```

Comment:
I don't believe we support symlinks in the plain bundle spec. I'm also not sure if git repositories can support symlinks on the remote. I guess it doesn't hurt to check but maybe it's adding some unnecessary complexity.

Reply:
I think git does support symlinks. If it is important to support them, we can do a follow-up when that time comes, IMO. There's some complexity with handling them if they point to something that would be outside the tar.

So for now, I opted to just ignore them.

Reply:
Yes, I think that's fair